runtime/cmd/decode-state-values/main.go

```diff
@@ -56,6 +60,11 @@ func worker(jobs <-chan entry, wg *sync.WaitGroup, decoded *uint64) {
 	var err error
 	var data []byte
 
+	decodeFunction := interpreter.DecodeValue
+	if *oldFlag {
+		decodeFunction = interpreter.DecodeValueV3
+	}
```

Comment:
Just an idea for consideration (not a proposal) -- what do you think about using version returned by `StripMagic` inside the loop that follows this?  The version returned can be used to pick the decode function.

Reply:
Good idea! I replaced it in 390c477898e9b4b295a1e830c93e0266b0c60c71